sycl/test-e2e/AddressSanitizer/nullpointer/nullpointer.cpp

```diff
@@ -0,0 +1,26 @@
+// REQUIRES: linux
+// RUN: %{build} %device_asan_flags -O0 -g -o %t
+// RUN: %{run} not %t 2>&1 | FileCheck %s
+// RUN: %{build} %device_asan_flags -O1 -g -o %t
+// RUN: %{run} not %t 2>&1 | FileCheck %s
+// RUN: %{build} %device_asan_flags -O2 -g -o %t
+// RUN: %{run} not %t 2>&1 | FileCheck %s
+
+#include <sycl/detail/core.hpp>
+
+int main() {
+  sycl::queue Q;
+  constexpr std::size_t N = 4;
+  int *array = nullptr;
+
+  Q.submit([&](sycl::handler &h) {
+    h.parallel_for<class MyKernel>(
+        sycl::nd_range<1>(N, 1), [=](sycl::nd_item<1> item) { array[0] = 0; });
+    Q.wait();
+  });
+  // CHECK: ERROR: DeviceSanitizer: null-pointer-access on Unknown Memory
```

Comment:
Can we construct cases to check null ptr for different address space?

Reply:
If I remember correctly, "0" is valid for local memory, and maybe we need to handle private memory as well.

Reply:
Test case for private memory was added, but can't enable now due to gfx driver issue